src/main/java/app/freerouting/gui/MainApplication.java

```diff
@@ -369,45 +370,78 @@ private void ExportBoardToFile(String filename) {
 
         // Get the correct panel from the option pane
         JPanel optionPanel = null;
+        int startButtonIndex = -1;
+        int cancelButtonIndex = -1;
         for(Component c : optionPane.getComponents())
           // Ensure there are buttons on the panel
-          if(c instanceof JPanel && ((JPanel)c).getComponents()[0] instanceof JButton){
+          if(c instanceof JPanel && ((JPanel)c).getComponents()[0] instanceof JButton && ((JPanel)c).getComponents()[1] instanceof JButton){
```

Comment:
Should make sure that there is more than one component packed into the `JPanel` first, something like:

```java
if(
  c instanceof JPanel &&
  ((JPanel)c).getComponents().length >= 2 && // Ensure that there is at least two components
  ((JPanel)c).getComponents()[0] instanceof JButton &&
  ((JPanel)c).getComponents()[1] instanceof JButton
){
```

Reply:
I agree 👍